pgrx/src/pgbox.rs

```diff
@@ -368,15 +369,64 @@ impl<T, AllocatedBy: WhoAllocated> PgBox<T, AllocatedBy> {
     }
 }
 
+impl<T> Clone for PgBox<T, AllocatedByPostgres>
+where
+    T: Clone,
+{
+    /// Copies the wrapped `T` into [`PgMemoryContexts::CurrentMemoryContext`].
+    fn clone(&self) -> Self {
+        unsafe {
+            let copy = PgMemoryContexts::CurrentMemoryContext
+                .copy_ptr_into(self.as_ptr(), std::mem::size_of::<T>());
+
+            PgBox::from_pg(copy)
```

Comment:
VERY uncertain about this.

Reply:
To elaborate: pgrx has made wrong assumptions in the past that T is all that is interesting about T, i.e. that the type is "flat" and not e.g. managing memory held elsewhere. This does something that could extend the lifespan of T but fail to extend the life of the memory that T is managing.

Reply:
I can get rid of this entirely.  Like the AsRef impl, this was just to make the roundtrip tests easier to write.

Maybe what we can do is `impl Clone for PgBox<pg_sys::Point>` and do that the "RightWay" rather than a blanket impl that, as you point out, probably ain't gonna work for every situation.

Thoughts?

Reply:
This needs to not only do this shallow copy of the memory that PgBox handles, it needs to do a "deep clone", so that if T is, say, `Vec<Box<[i32]>>`, it clones all the boxed slices too.

Reply:
Sure! This actually can work fine if it restricts to T being a Copy type.

Reply:
> Sure! This actually can work fine as-is if it restricts to T being a Copy type.

Yeah, I think maybe that was my intention anyways -- but obviously missed the mark.